### DISCUSSION TITLE
How can you run locally a github.io page?

### DISCUSSION THREAD

**Discussion by root2185** [community/community#37669] [Other Feature Feedback, Questions, & Ideas] [answered]
I cloned this github page  [pippim](https://github.com/pippim/pippim.github.io) and I put it on local host with apache2.
If I copy index.md to index.html it loads but it is really unusable.
This is how it should be: https://www.pippim.com/.

**Comment by airtower-luna** [comment]
That site seems to use the default Pages build, which uses [Jekyll](https://jekyllrb.com/) to build HTML from Markdown (among other things). So running Jekyll on the cloned data should work.

However, note:

* GitHub Pages uses a specific Jekyll version ([see the build log](https://github.com/pippim/pippim.github.io/actions/runs/3342279029/jobs/5534344781#step:4:16)). Building with a different (current) version may or may not work.
* Sites *can* define their own build workflow, so don't assume *all* Pages sites you see use Jekyll. Check if they have an Actions workflow that deploys to Pages.

**Reply by root2185** [reply]
Do you know to how set up `jekill` for the default Pages build?
Not sure if this is the case but I guess it is a good start, I asked.

`Pull down action image 'ghcr.io/actions/jekyll-build-pages:v1.0.4'`
Is that the `jekill` version?

**Reply by airtower-luna** [reply]
From the documentation in [`actions/jekyll-build-pages`](https://github.com/actions/jekyll-build-pages) that container image *seems* to be used for building pages sites, but I haven't used it myself. It's definitely not the same as installing Jekyll, but could be a good starting point if you want to exactly replicate the build.

However, I wonder what you're trying to achieve. If you just want a local copy of the finished site, replicating the build makes sense. On the other hand, if you want to build your own site it's probably better to start with a current Jekyll version (4.x) and fix any issues you encounter, so you don't start with a legacy tech stack.

**Reply by root2185** [reply]
I just want a local copy of that finished site.

**Comment by jfullstackdev** [comment]
it should be as simple as this 

https://jekyllrb.com/docs/

like any other local development setup

**Reply by root2185** [reply]
It couldn't start because of the missing `jekyll-theme-cayman`.

I compiled it following https://github.com/pietromenna/jekyll-cayman-theme/#how-to-use-it.
Even if I made it inside of one of the gem path (`gem env`) it couldn't find it.

Then I built it inside the pippim folder, so some files were overwritten.

Then it was missing `/_includes/head-custom-google-analytics.html`, I downloaded it from https://github.com/pages-themes/cayman/blob/master/_includes/head-custom-google-analytics.html.

In `assets/css/style.scss` I had to remove all the `@include`.

**Reply by jfullstackdev** [reply]
there are too many things involve, 

but let's have the themes, they are available, as documented here

https://jekyllrb.com/docs/themes/

also, I'm thinking why you need to run that locally when you can run that on GitHub Pages, right ? you need some staging ? there were posts in the past requesting that because as far as I can remember, they could not get correctly the proper setup in their local, so they need to have the staging on GitHub Pages so when they publish it, then it's all correct

unfortunately, as far as I know, that was not implement and not being implemented, not very sure if they have that in their roadmap

the only solution for that, as I can see is just simply host the project right there on GitHub Pages, maybe you can have another repo for staging and another repo for the production

another one is Codespace but I'm not very sure of the project whether it's available for Codespace setup

**Reply by root2185** [reply]
https://jekyllrb.com/docs/themes/#installing-a-theme
Editing the Gemfile of a new empty site and then adding the repo, overwriting the files, solved all the other problems.

I'd like to run this site completely offline, as a standalone app.
Furthermore this site is under development and may change exactly when you are using it.

**Reply by jfullstackdev** [reply]
nice, you were able to solve that, does it mean, you need the correct setup in Jekyll first before adding the repo ? so as to have a correct setup, that's what I can think of, similar to other situations

**Reply by root2185** [reply]
1. Make a new empty site `jekyll new <site>` 
2. Edit the Gemfile to use another theme: replace `gem "minima", "~> 2.5"` with `gem "jekyll-theme-<theme>"`.
3. Install the theme: `bundle install`
4. Create a Gemfile in the repo it needs only the theme line `gem "jekyll-theme-<theme>"`.
5. Run  the site `bundle exec jekyll serve`

But some files cannot be loaded: Task ending sounds on https://www.pippim.com/programs/tim-ta.html.
I selected instead a custom one that I put in the same folder and it works.

**Comment by root2185** [comment, marked as answer]
I found out that there is a guide but it doesn't explain to how make a Gemfile: https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll.

I used `gem "github-pages", group: :jekyll_plugins` ispired by https://talk.jekyllrb.com/t/how-to-get-a-cloned-github-pages-repo-to-run-locally/6240.

I also tried `gem "github-pages", "~> 227", group: :jekyll_plugins` seeing step 10 on https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll#creating-your-site.

The result is however always as my reply in the other answer.

But the last step fixes the missing files, `bundle exec jekyll serve` will show `http://127.0.0.1:4000` but you have to use localhost instead (see below).
And this method should be the better:

1. Clone the repo.
2. Create a Gemfile with this content: 
```
source "https://rubygems.org"
gem "github-pages", group: :jekyll_plugins
```
4. Run `bundle install`.
5. Run your Jekyll site locally `bundle exec jekyll serve`. 
6. Connet to `http://localhost:4000` instead of `http://127.0.0.1:4000/`.
(It shows this and it is also a link `Server address: http://127.0.0.1:4000`)

**Reply by mohsin651** [reply]
Thanks man! It worked.

**Reply by sahilrajput03** [reply]
Thanks man! It does work. I had to make my Gemfile like this:

```Gemfile
source "https://rubygems.org"
# Note: 231 version specified below comes from this link - https://pages.github.com/versions/
gem "github-pages", "~> 231", group: :jekyll_plugins
gem "webrick", "~> 1.8"
```

**Reply by hoile9119** [reply]
thanks, help me out!

**Comment by Lanrad23** [comment]
If you encounter errors during the bundle install step, make sure your Ruby and Bundler installations are correct and up to date. You might also need to install development tools for Ruby, depending on your operating system.
Also, ensure the repository includes a Gemfile specifying Jekyll and any necessary plugins. If not, you may need to set up Jekyll manually for the site, which involves creating a new Jekyll site and moving the content over.
